scripts/zones/The_Shrine_of_RuAvitau/mobs/Seiryu.lua

```diff
@@ -2,31 +2,33 @@
 -- Area: The Shrine of Ru'Avitau
 --  Mob: Seiryu (Pet version)
 -----------------------------------
+mixins = { require('scripts/mixins/job_special') }
+-----------------------------------
 ---@type TMobEntity
 local entity = {}
 
 entity.onMobSpawn = function(mob)
+    mob:setMobMod(xi.mobMod.ADD_EFFECT, 1)
     mob:setMobMod(xi.mobMod.CANNOT_GUARD, 1)
-end
+    mob:setMod(xi.mod.REGAIN, 450)
 
-entity.onMobMagicPrepare = function(mob, target, spellId)
-    if not mob:hasStatusEffect(xi.effect.HUNDRED_FISTS, 0) then
-        local rnd = math.random(1, 100)
-        if rnd <= 50 then
-            return 186 -- aeroga 3
-        elseif rnd <= 70 then
-            return 157 -- aero 4
-        elseif rnd <= 90 then
-            return 208 -- tornado
-        else
-            return 237 -- choke
+    mob:addListener('EFFECT_LOSE', 'SEIRYU_HF', function(mobArg, effect)
+        if effect:getEffectType() == xi.effect.HUNDRED_FISTS then
+            mobArg:setMagicCastingEnabled(true)
+            mobArg:setMobAbilityEnabled(true)
         end
-    end
+    end)
+end
 
-    return 0 -- Still need a return, so use 0 when not casting
+entity.onAdditionalEffect = function(mob, target, damage)
+    return xi.mob.onAddEffect(mob, target, damage, xi.mob.ae.ENAERO)
 end
 
-entity.onMobDeath = function(mob, player, optParams)
+entity.onMobWeaponSkill = function(target, mob, skill)
+    if skill:getID() == xi.jsa.HUNDRED_FISTS then
+        mob:setMagicCastingEnabled(false)
+        mob:setMobAbilityEnabled(false)
+    end
```

Comment:
I'm starting to wonder if it wouldnt just be easier to move this logic to the actual effect script, instead of needing a listener and WS-use logic for every 2-hour using mob?